app_feup/lib/view/Widgets/generic_card.dart

```diff
@@ -36,9 +36,9 @@ abstract class GenericCard extends StatefulWidget {
     return Container(
         child: Text(
             'última atualização às ' +
-                t.hour.toString() +
+                t.hour.toString().padLeft(2, '0') +
                 ':' +
-                t.minute.toString(),
+                t.minute.toString().padLeft(2, '0'),
```

Comment:
I can see that there are other places, such as the `schedule_card`, that share the same logic of using `padLeft`. What about creating an extension method, maybe something called like `toTimeString` that automatically converts an `int` to a time-like string? 

Here's some more details on extension methods: https://dart.dev/guides/language/extension-methods

This would also require us to update the `pubspec.yaml` file to use sdk 2.7.0 but I think that's healthy anyways (hopefully nothing breaks!)

Reply:
Don't forget to update the `pubspec.yaml` to use at the very least the 2.7.0 version of dart sdk, instead of the current 2.2.0. Also, kindly update the version of the app of the file as well!

Reply:
Thanks! I've changed the sdk version to 2.7.0, `sdk: ">=2.7.0 <3.0.0"`, and the app version: `version: 1.0.0+8`. Not sure if this is correct though, and would appreciate a review on this changes. Meanwhile running the process, the program crashed.

Reply:
I don't see any issues in this and upgrading should be done flutter's inteface by calling `flutter pub get`. Wdym by the process crashing?

Reply:
Now, it's working perfectly, by running the right command. Thank you!